fix(vcpkg): database_system port build fails (libpq/postgres_manager) in ecosystem integration test

## What

The ecosystem integration test workflow (PR #608, issue #604) discovered that the `database_system` vcpkg port fails to build on both platforms:

- **macOS**: `libpq:arm64-osx` build failure — `CMAKE_MAKE_PROGRAM` and `CMAKE_CXX_COMPILER` not set
- **Ubuntu**: `postgres_manager.cpp` compilation failure in `kcenon-database-system:x64-linux`

## Where

- **Workflow**: `.github/workflows/ecosystem-vcpkg-integration.yml` — Layer 5
- **Platforms**: ubuntu-24.04, macos-14

## Evidence

Failed CI run: https://github.com/kcenon/common_system/actions/runs/24074150510

macOS error:
```
error: building libpq:arm64-osx failed with: BUILD_FAILED
CMake Error: CMAKE_MAKE_PROGRAM is not set
CMake Error: CMAKE_CXX_COMPILER not set, after EnableLanguage
```

Ubuntu error:
```
FAILED: database/CMakeFiles/database.dir/postgres_manager.cpp.o
error: building kcenon-database-system:x64-linux failed with: BUILD_FAILED
```

## Root Cause

- macOS: The `libpq` (PostgreSQL client library) port may need `autoconf`/`make` tools that aren't available, or the vcpkg triplet configuration is incomplete
- Ubuntu: Compilation error in `postgres_manager.cpp` — possibly a missing header, API change, or dependency version mismatch

## Acceptance Criteria

- [ ] `database_system` installs successfully via vcpkg on ubuntu-24.04
- [ ] `database_system` installs successfully via vcpkg on macos-14
- [ ] Ecosystem integration test Layer 5 passes on both platforms

## Context

Discovered by the ecosystem verification gate (#604). This is a remediation issue.

--- a/tests/ecosystem-consumer/database-system/vcpkg.json
+++ b/tests/ecosystem-consumer/database-system/vcpkg.json
@@ -1,5 +1,10 @@
 {
   "name": "test-database-system",
   "version-string": "0.0.0",
-  "dependencies": ["kcenon-database-system"]
+  "dependencies": [
+    {
+      "name": "kcenon-database-system",
+      "default-features": false
+    }
+  ]
 }

